packages/server/mailing/mailing.schema.js

```diff
@@ -149,9 +149,118 @@ MailingSchema.methods.duplicate = function duplicate(_user) {
 
   return this;
 };
+MailingSchema.index({ createdAt: 1 });
+MailingSchema.index({ createdAt: -1 });
+MailingSchema.index({ updatedAt: 1 });
+MailingSchema.index({ updatedAt: -1 });
+MailingSchema.index({ name: 1 });
+MailingSchema.index({ name: -1 });
+MailingSchema.index({ author: 1 });
+MailingSchema.index({ author: -1 });
+MailingSchema.index({ wireframe: 1 });
+MailingSchema.index({ wireframe: -1 });
+MailingSchema.index({ tags: 1 });
+MailingSchema.index({ tags: -1 });
 
 MailingSchema.statics.findForApi = async function findForApi(query = {}) {
-  return this.find(query, { previewHtml: 0, data: 0 });
+  const { paginationJSON, filtersJSON, ...restQuery } = query;
+  const additionalQueryParams = {};
+
+  if (paginationJSON && paginationJSON.page && paginationJSON.itemsPerPage) {
+    additionalQueryParams.page = paginationJSON.page;
+    additionalQueryParams.limit = parseInt(paginationJSON.itemsPerPage);
+  }
+
+  if (
+    paginationJSON.sortBy &&
+    Array.isArray(paginationJSON.sortBy) &&
+    paginationJSON.sortDesc &&
+    Array.isArray(paginationJSON.sortDesc)
```

Comment:
```suggestion
    Array.isArray(paginationJSON.sortBy) &&
    Array.isArray(paginationJSON.sortDesc)
```

because `Array.isArray(null) === Array.isArray(undefined) === false` ;)
<img width="261" alt="CleanShot 2022-01-20 at 10 10 11@2x" src="https://user-images.githubusercontent.com/7521065/150308281-b63f4648-1a70-44c1-9e0b-b2c44f7397c5.png">